tide-whisperer.go

```diff
@@ -99,6 +117,23 @@ func main() {
 		return !(perms["root"] == nil && perms["view"] == nil)
 	}
 
+	//log error detail and write as application/json
+	jsonError := func(res http.ResponseWriter, err detailedError, startedAt time.Time) {
+
+		if err.RawMessage == "" {
```

Comment:
You can probably get rid of this and just use "omitempty" on the RawMessage json serialization.  No need to add duplicate data to the logs.